crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -462,7 +462,8 @@ where
     ) -> bool {
         // On Optimism, the proposers are allowed to reorg their own chain at will.
         #[cfg(feature = "optimism")]
-        if self.blockchain.chain_spec().is_optimism() {
+        if reth_chainspec::EthChainSpec::chain(self.blockchain.chain_spec().as_ref()).is_optimism()
```

Comment:
unsure what's the best way to do this. for now just changed to checks on chain id, but I guess there are cases that we don't cover by this?

Reply:
We prob should get rid of the feature flag altogether and capture this logic in the OP engine consensus impl?